### ISSUE TITLE
Lerna, monorepos, and package-specific hooks

### ISSUE THREAD

**Issue by ELLIOTTCABLE** [typicode/husky#624]
At the moment, in a Lerna-style monorepo, I can only add Husky hooks *in the root project*. That is, if Husky is installed in `root/node_modules`, and a version specified in `root/package.json`, then Husky installs its hooks in `root/.git/hooks` (all fine) … but only examines `root/.huskyrc.json` for commands.

Meanwhile, in the projects I'm currently combining into a monorepo, I have individual Husky-tasks that are project-specific. I initially tried to keep these in `packages/a_module/.huskyrc.json`; but Husky doesn't seem to recognize them there:

```json5
// .../root/packages/a_module/.huskyrc.json
{
   "hooks": {
      "pre-commit": "lint-staged && run-p test:js:staged test:ml"
   }
}
```

```sh
$ git commit
husky:debug husky v3.1.0 (created at 1/1/2020, 10:10:24 PM)
husky:debug pre-commit hook started
husky:debug Current working directory is '.../root'
husky:debug Changed current working directory to '.../root'
husky:debug husky v3.1.0 (created at 1/1/2020, 10:10:24 PM)
husky:debug prepare-commit-msg hook started
husky:debug Current working directory is '.../root'
husky:debug Changed current working directory to '.../root'
Aborting commit due to empty commit message.
```

One way around this, is to simply revert to pre-v1.0.0-style hooks: I can add an `npm script` for every single hook, to every single package …

```json5
// .../root/packages/a_module/package.json
{
   "scripts": {
      "pre-commit": "lint-staged && run-p test:js:staged test:ml",
   }
}
```

… and then the root-level `root/.huskyrc.json` can instead specify:

```json5
// .../root/.huskyrc.json
{
   "hooks": {
      "pre-commit": "npm run precommit"
   }
}
```

However, I would really strongly-prefer not to couple the `npm scripts` and my CI/precommit infrastructure that tightly, and/or further complicate my already-vast `npm scripts` list, heh.

**tl;dr**: It'd be extremely nice if Husky would look at `.huskyrc` files _in any subdirectory with a `package.json`_, especially in a Lerna-configured environment!

**Comment by typicode** [comment]
I haven't tried it, but what about something like this?

`root/.huskyrc.json`
```json
"pre-commit": "lerna run precommit --since ..."
```

`root/packages/A/package.json`
```json
"scripts": {
  "precommit": "<specific cmd to A>"
}
```

`root/packages/B/package.json`
```json
"scripts": {
  "precommit": "<specific cmd to B>"
}
```

https://github.com/lerna/lerna/tree/master/core/filter-options#--since-ref

This should run `precommit` scripts from packages that has changed `--since ...` and you can define specific commands too.

**Comment by ELLIOTTCABLE** [comment]
Hm. I'm not sure how `--since` helps, too much here.

Also, FYI, this isn't a blocker for me — it's what I saw as a slight deficiency, and wanted to get on your radar; but for me, it's actually `lint-staged` that I badly need to support multiple-subdirectory-configuration. That said, invoking `lerna run` in the root project faces a lot of the same issues that I [explain in fair detail over here](https://github.com/okonet/lint-staged/issues/760#issuecomment-573947384)!

Feel free to just throw this on a todo list somewhere and not worry too much about it, unless somebody else also comes along begging. (=

**Comment by typicode** [comment]
Feedback is always appreciated thanks :)
